### PR TITLE
fix: output logs for all execa commands

### PR DESCRIPTION
**Pull request by dansaadati**
This changes ensures that logs are output from the child process that execa produces.

### CODE REVIEW

File: lib/util.mjs [hunk 1]
```diff
@@ -56,3 +57,21 @@ export function replace(values, valueProp, replacerMap) {
     }
   });
 }
+
+export async function execaNodeAndLog(cmd, args, cwd) {
```
Comment (tbpg) [inline lib/util.mjs:61]: Could we do something like this?

```
export function withLog(fn) {
  return (cmd, args, cwd) => {
    const opts = {cwd};

    opts.stdout = process.stdout;
    opts.stderr = process.stderr;

    return fn(cmd, args, opts);
  }
}
```

Then call it with something like:

```
withLog(execa)(cmd, args, cwd);
withLog(execaNode)(cmd, args, cwd);
```

Fine if you prefer it like this for simplicity. There may also be a better way to generalize.

Comment (dansaadati) [reply lib/util.mjs]: Good point! I added an optional argument to specify whether the command is meant to be run on Node or not, which allows us to use just one function.

File: lib/util.mjs [hunk 2]
```diff
@@ -56,3 +57,25 @@ export function replace(values, valueProp, replacerMap) {
...
+/**
+ * Executes command in a subprocess.
+ *
+ * @param {string} cmd The command to run in the subprocess.
+ * @param {Array<string>} args Additional args to include when running the command.
+ * @param {string} cwd The working directory in which to run the command.
+ * @param {boolean=} isNodeCmd If set, command will be run in Node.
+ * @returns A Promise containing the result of the command.
+ */
+export async function execaAndLog(cmd, args, cwd, isNodeCmd = false) {
```
Comment (tbpg) [inline lib/util.mjs:70]: Sorry, I think I actually preferred it before. It's more verbose, but I find the `isNodeCmd` parameter confusing. You have to do things like `/*isNodeCmd=*/true`.

Comment (dansaadati) [reply lib/util.mjs]: Understood! I think we could avoid the inline comment by accepting a destructured object as a parameter:
```export async function execaAndLog({cmd, args, cwd, isNodeCmd = false}) { ... }```
and then call the function like:
```execaAndLog({cmd=cmd, args=args, cwd=cwd,, isNodeCmd = false}) .```

However, that's not really a pattern we use in this codebase. Instead, I wrote a higher order function to call `execa` commands with logs.

**Review by tbpg — APPROVED** [review]

**Pull request merged**